pkg/auth/audit/audit.go

```diff
@@ -1,33 +1,46 @@
+// Package audit is used to preform audit logging.
 package audit
 
 import (
 	"bytes"
+	"compress/gzip"
+	"compress/zlib"
 	"context"
 	"encoding/json"
+	"fmt"
+	"io"
 	"io/ioutil"
 	"net/http"
 	"regexp"
 	"strings"
 	"time"
 
 	"github.com/pborman/uuid"
-	"github.com/pkg/errors"
 	v32 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
 	"github.com/sirupsen/logrus"
 	k8stypes "k8s.io/apimachinery/pkg/types"
 	"k8s.io/apiserver/pkg/endpoints/request"
 )
 
 const (
-	contentTypeJSON = "application/json"
-	redacted        = "[redacted]"
+	contentTypeJSON     = "application/json"
+	contentEncodingGZIP = "gzip"
+	contentEncodingZLib = "deflate"
+	redacted            = "[redacted]"
 )
 
+// Level represents a desired logging level
+type Level int
+
 const (
-	levelNull = iota
-	levelMetadata
-	levelRequest
-	levelRequestResponse
+	// LevelNull default value
+	LevelNull Level = iota
```

Comment:
Do these need to be exported? I'm not seeing where they are used outside of this package

Reply:
I changed the function signatures of `func NewLogWriter(path string, level Level, maxAge, maxBackup, maxSize int)` to use `audit.Level` instead of an arbitrary `int`. This does not change the functionality of anything, but it seemed odd that to create a new `LogWriter` you have to look into the package source code to know what each int value does.